vaadin-list-mixin.html

```diff
@@ -169,19 +174,18 @@
       const condition = item => !(item.disabled || this._isItemHidden(item));
       let idx, increment;
 
-      const isRTL = !this._vertical && this.getAttribute('dir') === 'rtl';
-      const dirIncrement = isRTL ? -1 : 1;
+      const dirIncrement = this._isRTL ? -1 : 1;
 
       if (this._vertical && key === 'Up' || !this._vertical && key === 'Left') {
         increment = -dirIncrement;
         idx = currentIdx - dirIncrement;
       } else if (this._vertical && key === 'Down' || !this._vertical && key === 'Right') {
         increment = dirIncrement;
         idx = currentIdx + dirIncrement;
-      } else if ((key === 'Home' && !isRTL) || (key === 'End' && isRTL)) {
+      } else if ((key === 'Home' && !this._isRTL) || (key === 'End' && this._isRTL)) {
         increment = 1;
         idx = 0;
-      } else if ((key === 'End' && !isRTL) || (key === 'Home' && isRTL)) {
+      } else if ((key === 'End' && !this._isRTL) || (key === 'Home' && this._isRTL)) {
         increment = -1;
         idx = this.items.length - 1;
       }
```

Comment:
This logic seems to introduce some behavior that might be a bit off. 

Below, I hit `Home` and then `End`. First it goes to the last item and then to the first one.
![list-rtl](https://user-images.githubusercontent.com/262432/77414687-1c123c80-6dca-11ea-866f-fc278787c03a.gif)

On the Macbook keyboard, there's no dedicated `Home` nor `End` buttons, but the user can hit either `Fn`+ ⬅️ or `Fn` + ➡️ and they act as these buttons.

While hitting the `Fn`+ ⬅️ or `Fn` + ➡️ combinations make sense because the direction visual feedback from the keys, hitting `Home`/`End` alone doesn't seem right from my perspective (not being a speaker of any RTL language). 

If that's the case, `Home`/`End` should work differently of `Fn`+ ⬅️ or `Fn` + ➡️ in RTL, then the trick part is that Mac OS doesn't differentiate the keys, so it's not possible to determine which button/combination was used.

Then, it might be enough to make it work differently for Mac OS and other platforms.

@mehdi-vaadin, could you please add your point of view here, please? Do you expect that the `Home`/`End` buttons to behave differently:
 - `Home`: goes to the right most element
 - `End`: goes to the left most element
 -  `Fn`+ ⬅️: goest to the left most element
 - `Fn` + ➡️: goes to the right most element

Reply:
`Home` and `End` should behave the same in both RTL and LTR. `Home` should show the first element and `End` should show the last element. Note that if you use an external keyboard for your MacBook, you may have separate `Home` and `End` keys. 

Fn+⬅️and Fn+➡️should show the left-most and the right-most elements no matter what the direction is.

Reply:
@mehdi-vaadin What is your assumption about Fn+arrow key based on (how it should work)? Is there some best practice or articles related to this or does it behave like that in some native macOS apps commonly (if using RTL)?

It seems that when you press `Fn` + left arrow on mac at least in a browser it triggers exactly the same keycodes/events as if pressing Home key (regardless of RTL or not) so it's impossible to detect in JS if the user pressed an actual "Home" key or the `Fn` + left arrow shortcut.

To me this seems like that's how macOS has chosen to implement it (so `Fn` + arrow left always simulates press of Home key) so it's not really up to component/app developer to change it.

Reply:
You can test the keycodes that are triggered by pressing keys while on this page: https://keycode.info/

Reply:
How does the cursor behave in a native `<input>` in HTML or a native text input in a macOS app in RTL mode when using the `Fn`+ left arrow for example? Does it always go to left also in RTL or not?

Reply:
Hmm, it behaves weirdly with native `input` on my machine: with `Fn` + :arrow_left:  ( Home ) it goes right, but with `Fn` + :arrow_right:  ( End ) it doesn’t move the cursor, but instead scrolls the page

Reply:
I tested it on Google Docs and it works as I mentioned above. I'm not sure how it's implemented. But, it works correctly. You can try it yourself by adding some Farsi text in an empty doc. Try the following text. 
دوستت دارم.

Reply:
Maybe it's something missing on my OS configuration, but when I printed the Farsi text you provided as example at Google Docs, hitting `Fn` + ⬅️ brought the cursor to the beginning of the text and the opposite happened when I pressed `Fn` + ➡️.

Reply:
Sorry, my mistake. You're right. I was using `Command` + Arrow keys.

Reply:
@Haprog you are right. There is no difference between `Fn`+⬅️ and `Home`. We can't distinguish them. I'd say both of them should show the first element.

Reply:
Thanks, @mehdi-vaadin 🙇🏻‍♂️

Reply:
Thank you, guys!
I have simplified the logic for `Home` and `End`. There is no difference now.